### DISCUSSION TITLE
Collabora: Failed to connect to the remote server - make inheritance of dns settings possible

### DISCUSSION THREAD

**Discussion by Mantany** [nextcloud/all-in-one#5019] [Questions] [answered]
Hi, I tried installing the aio-nextloud behind a traefik reverse proxy - I got it working so far.
I also have my dns server on that same docker host to resolve to a local domain. 

### Steps to reproduce
To make the aio container use the specific dns server, I used the following config (ansible):

name: nextcloud-aio-mastercontainer
        image: "nextcloud/all-in-one:latest"
        init: "true"
        restart_policy: always
        ports:
          - 11001:8080
        volumes:
          - /var/run/docker.sock:/var/run/docker.sock:ro
          - "nextcloud_aio_mastercontainer:/mnt/docker-aio-config"
        env:
          APACHE_PORT: "11000"
          APACHE_IP_BINDING: "0.0.0.0"
          NEXTCLOUD_DATADIR: "{{ nfs_mount }}/data/"
          NEXTCLOUD_MOUNT: "{{ nfs_mount }}/data"
          NEXTCLOUD_UPLOAD_LIMIT: 10G
          NEXTCLOUD_MAX_TIME: "3600"
          NEXTCLOUD_MEMORY_LIMIT: 1024M
          NEXTCLOUD_KEEP_DISABLED_APPS: "false"
        # Important: point to local dns server to resolve private dns domain
        **dns_servers:
          - "192.168.0.124"**

It seems that I can get past the domain check, and it can talk successfully with my reverse proxy. 

The error occurs, when collabora starts: 
Failed to connect to the remote server: cURL error 6: Could not resolve host

Logs nextcloud-aio-collabora:
wsd-00007-00018 2024-07-21 20:09:35.500529 +0200 [ remotefontconfig_poll ] ERR  Failed to lookup host [nextcloud.local.ltrippensee.de]. Skipping (0: Success)| net/NetUtil.cpp:364

wsd-00007-00018 2024-07-21 20:09:35.500593 +0200 [ remotefontconfig_poll ] ERR  #-1: Failed to connect to nextcloud.local.ltrippensee.de:443| net/HttpRequest.hpp:1320

wsd-00007-00018 2024-07-21 20:09:35.500630 +0200 [ remotefontconfig_poll ] ERR  Remote config server has response status code: 0 (Unknown)| wsd/COOLWSD.cpp:1207

I found, that the collabora container doesn't inherits the dns_servers and dns_search_domains settings from the mastercontainer.

From Inspect nextcloud-aio-mastercontainer:
"HostConfig": {
...
 "Dns": [
            "192.168.0.124"
        ],
        "DnsOptions": null,
        "DnsSearch": [
            "*.local.example.de"
        ],
}

From Inspect nextcloud-aio-collabora:
"HostConfig": {
...
"Dns": null,
        "DnsOptions": null,
        "DnsSearch": null,
}

If I manually change the dns setting of that container, I can reach the the nextcloud container.

### Expected behavior
Inheritance of dns_servers settings for child /managed containers.

### Actual behavior
No inheritance of the dns_servers settings for child containers. You are therefore forced to change the dns settings host-wide. 
Even if you do not want to route all of the host's traffic through the dns server.

### Host OS
Ubuntu 24.04 LTS

#### Nextcloud AIO version
v9.2.0
#### Current channel
stable

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one#how-can-i-access-nextcloud-locally

**Comment by Mantany** [comment]
Hi, thank you for the link - I have read the docs - it seems that the solution it to either:
- point to the custom dns server network wide (e.g. router settings)
- changing the docker config on the whole host: by editing e.g. /etc/docker/daemon.json and adding { "dns" : [ "ip.address.of.this.server" , "8.8.8.8" ] }  

If im not mistaken: there is no way (currently) to point just for the specific containers to the dns server? 
e.g. by using the dns_servers settings in the docker container itself. 

My reasoning is that I don't want a whole docker host to route all of the host's traffic through the dns server.
